### PR TITLE
Import StrictVersion and LooseVersion from setuptools.distutils.verison or setuptools._distutils.version

### LINKED ISSUE #63350: [BUG] setuptools._distutils not available on Debian 10 (setuptools v40.8.0)
**Description**
tests.integration.ssh.test_state.SSHStateTest.test_show_highstate (from pytest) fails due to 
```
from setuptools._distutils.version import LooseVersion as _LooseVersion\nModuleNotFoundError: No module named \'setuptools._distutils\'
```

**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info. There is no general set-up of Salt.)
Testing framework for Debian 10.
```
[Dashboard](https://jenkins.saltproject.io/)
[Salt PR Debian 10 amd64 Py3 Pytest Builds](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/)
[master](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/)
[#2776](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/)
[Test Results](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/)
[Test Results](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/)
[tests.integration.ssh.test_state](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/tests.integration.ssh.test_state/)
[SSHStateTest](https://jenkins.saltproject.io/job/pr-debian-10-amd64-py3-pytest/job/master/2776/testReport/junit/tests.integration.ssh.test_state/SSHStateTest/)
test_show_highstate
```

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [X] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD
- [ ] classic packaging
- [X] onedir packaging
- [ ] used bootstrap to install

**Steps to Reproduce the behavior**
Run test above

**Expected behavior**
Test to pass

**Screenshots**

```
self = <tests.integration.ssh.test_state.SSHStateTest testMethod=test_show_highstate>

    def tearDown(self):
        """
        make sure to clean up any old ssh directories
        """
        salt_dir = self.run_function("config.get", ["thin_dir"], wipe=False)
>       self.assertIsInstance(salt_dir, (str,))
E       AssertionError: {'_error': 'Failed to return clean data', 'stderr': 'Traceback (most recent call last):\n  File "/var/tmp/.root_0e6fa3_salt/salt-call", line 26, in <module>\n    from salt.scripts import salt_call\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/__init__.py", line 146, in <module>\n    import salt._logging  # isort:skip\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/__init__.py", line 12, in <module>\n    from salt._logging.impl import (\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/impl.py", line 28, in <module>\n    from salt._logging.handlers import DeferredStreamHandler  # isort:skip\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/_logging/handlers.py", line 16, in <module>\n    from salt.utils.versions import warn_until_date\n  File "/var/tmp/.root_0e6fa3_salt/pyall/salt/utils/versions.py", line 21, in <module>\n    from setuptools._distutils.version import LooseVersion as _LooseVersion\nModuleNotFoundError: No module named \'setuptools._distutils\'', 'stdout': '', 'retcode': 1} is not an instance of (<class 'str'>,)
```

**Versions Report**
relenv testing with master branch

</details>

**Additional context**
Wondering how the original PR https://github.com/saltstack/salt/pull/63309 with the changes for setuptools._distutils.version passed successfully, or was reported as successful.

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Import StrictVersion and LooseVersion from setuptools.distutils.verison or setuptools._distutils.version if the first attempt to import fails, thus allow for old and newer versions of setuptools.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/63350

### Previous Behavior
Test would fail on Debian 10, due to ImportError, since setuptools._distutils is not available in setuptools v40.8.0 which is current for Debian 10

### New Behavior
Works with new and old versions of setuptools, and tests pass.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by s0undt3ch** [comment]
re-run full all

File: salt/utils/versions.py
Comment (s0undt3ch) [inline salt/utils/versions.py:46]:
```suggestion
try:
    from setuptools.distutils.version import LooseVersion as _LooseVersion
    from setuptools.distutils.version import StrictVersion as _StrictVersion
except ImportError:
    from setuptools._distutils.version import LooseVersion as _LooseVersion
    from setuptools._distutils.version import StrictVersion as _StrictVersion
```
Let's just try/catch both alternatives.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/versions.py
Comment (s0undt3ch) [reply salt/utils/versions.py:46]: And because of the ssh errors, a third fallback into importing from distutils, since the looseversion package is not being installed.

**Comment by s0undt3ch** [comment]
This PR needs to go through the full test suite.

File: salt/utils/versions.py
Comment (dmurphy18) [reply salt/utils/versions.py:46]: Wanted to isolate the import so there was individual log messages. I can combine them and just have the one generic log message, as there should be a log message imho.

**Comment by dmurphy18** [comment]
@s0undt3ch Actually when I checked late last night it had gone through the full test suite, or whatever test suites we have for PRs with no problem.  Which raises the question, why are they failing now ?.
Also the original PR also made it through the full PR test suite, so why didn't they catch the issue then ?

Wondering what holes we have in testing to explain the inconsistencies ?

File: salt/utils/versions.py
Comment (s0undt3ch) [reply salt/utils/versions.py:46]: Don't log import errors, unless there's some usefulness to that.
Also don't raise `ImportError()`, raise the original exception.

Comment (dmurphy18) [reply salt/utils/versions.py:46]: Updated to similar with log.debug messages

File: salt/utils/versions.py
Comment (s0undt3ch) [inline salt/utils/versions.py]:
```suggestion
# pylint: disable=blacklisted-module
try:
    from setuptools._distutils.version import LooseVersion as _LooseVersion
    from setuptools._distutils.version import StrictVersion as _StrictVersion
except ImportError:
    try:
        from setuptools.distutils.version import LooseVersion as _LooseVersion
        from setuptools.distutils.version import StrictVersion as _StrictVersion
    except ImportError:
        from distutils.version import LooseVersion as _LooseVersion
        from distutils.version import StrictVersion as _StrictVersion

# pylint: enable=blacklisted-module
import salt.version

log = logging.getLogger(__name__)
```

**Comment by dmurphy18** [comment]
re-run full all

File: salt/utils/pkg/win.py
Comment (s0undt3ch) [inline salt/utils/pkg/win.py]:
```suggestion
from salt.utils.versions import LooseVersion
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: salt/utils/versions.py
Comment (dmurphy18) [reply salt/utils/versions.py]: Just updated and submitted, can you look at recent code

File: salt/utils/pkg/win.py
Comment (dmurphy18) [reply salt/utils/pkg/win.py]: Just updated and submitted, can you look at recent code

File: salt/utils/versions.py
Comment (s0undt3ch) [reply salt/utils/versions.py]: > I believe logging is important

In this particular scenario, will fallbacks in place, it's not.

File: salt/utils/pkg/win.py
Comment (dmurphy18) [reply salt/utils/pkg/win.py]: done

File: salt/utils/versions.py
Comment (dmurphy18) [reply salt/utils/versions.py]: okay logging is gone, and implemented as suggested

**Comment by dmurphy18** [comment]
re-run full all

**Review by s0undt3ch — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Comment by waynew** [comment]
When this gets merged, can someone update https://github.com/saltstack/salt/pull/63257 ?

That should get all of the tests passing there 👍

**Comment by dmurphy18** [comment]
I feel like I am 'Waiting for Godot' these two tests, RHEL 7 & 9 are taking forever, over two hours now.
@waynew can you review and approve, if find nothing, so have 3 approvals when the tests finally finish

**Pull request merged**
